addons/web/static/src/views/form/form_controller.js

```diff
@@ -161,9 +161,7 @@ export class FormController extends Component {
                     return this.model.root.save({ noReload: true, stayInEdition: true });
                 }
             },
-            beforeUnload: () => {
-                return this.model.root.urgentSave();
-            },
+            beforeUnload: this.beforeUnload,
```

Comment:
I don't think this can work (`this` won't be bound)
```suggestion
            beforeUnload: () => this.beforeUnload(),
```